src/assistant.js

```diff
@@ -213,19 +219,22 @@ export async function openAssistantForm({
 			view.selectedTaskId = null
 			lastTask = null
 		})
-		view.$on('background-notify', () => {
-			cancelTaskPolling()
-			view.showScheduleConfirmation = true
-			view.showSyncTaskRunning = false
-			view.loading = false
-			setNotifyReady(lastTask.id)
+		view.$on('background-notify', (enable) => {
+			setNotifyReady(lastTask.id, enable).then(res => {
+				view.isNotifyEnabled = enable
+			})
 		})
 		view.$on('cancel-task', () => {
 			cancelTaskPolling()
-			cancelTask(lastTask.id)
-			view.showSyncTaskRunning = false
-			view.loading = false
-			lastTask = null
+			setNotifyReady(lastTask.id, false).finally(res => {
+				view.isNotifyEnabled = false
+				cancelTask(lastTask.id).then(res => {
+					view.showSyncTaskRunning = false
+					view.loading = false
+					view.selectedTaskId = null
+					lastTask = null
+				})
+			})
 		})
```

Comment:
```suggestion
			setNotifyReady(lastTask.id, false)
			cancelTask(lastTask.id).then(res => {
				view.showSyncTaskRunning = false
				view.loading = false
				view.selectedTaskId = null
				lastTask = null
			})
		})
```

We don't need to wait for setNotifyReady to be done and then cancel the task. We can do the requests concurrently.